OMCompiler/Compiler/NBackEnd/Util/NBDifferentiate.mo

```diff
@@ -762,13 +762,20 @@ public
       local
         Expression ret;
 
-      // all these are integer based and therefore zero
-      case ("abs")      then Expression.makeZero(Type.INTEGER());
-      case ("sign")     then Expression.makeZero(Type.INTEGER());
-      case ("ceil")     then Expression.makeZero(Type.INTEGER());
-      case ("floor")    then Expression.makeZero(Type.INTEGER());
+      // all these have integer values and therefore zero derivative
+      case ("sign")     then Expression.makeZero(Expression.typeOf(arg));
+      case ("ceil")     then Expression.makeZero(Type.REAL());
+      case ("floor")    then Expression.makeZero(Type.REAL());
```

Comment:
Is this correct? the return type of `floor` etc. is integer, so the derivative is also of type zero right?

Reply:
No, the return type is Real, see https://specification.modelica.org/maint/3.5/operators-and-expressions.html#modelica:floor